internal/accrual/api/client/client.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/northmule/gophermart/internal/app/services/logger"
+	"io"
+	"net/http"
+	"strings"
+)
+
+const ServiceAccrualURI = "/api/orders/{number}"
+const ServiceName = "Accrual"
+
+type ErrorNoContent string
+type ErrorTooManyRequests string
+type ErrorInternalServerError string
+type ErrorUndefined struct {
+	code int
+}
+
+func (err ErrorNoContent) Error() string {
+	return fmt.Sprintf("сервис %s вернул пустой ответ", ServiceName)
+}
+
+func (err ErrorTooManyRequests) Error() string {
+	return fmt.Sprintf("сервис %s перегружен запросами", ServiceName)
+}
+
+func (err ErrorInternalServerError) Error() string {
+	return fmt.Sprintf("сервис %s сломался", ServiceName)
+}
+
+func (err ErrorUndefined) Error() string {
+	return fmt.Sprintf("сервис %s вернул не обработаный код ошибки %d", ServiceName, err.code)
+}
+
+type AccrualClient struct {
+	serviceURL string
+	logger     *logger.Logger
+}
+
+func NewAccrualClient(serviceURL string, logger *logger.Logger) *AccrualClient {
+	instance := &AccrualClient{
+		serviceURL: serviceURL,
+		logger:     logger,
+	}
+	return instance
+}
+
+type ResponseAccrual struct {
+	Order   string  `json:"order"`
+	Status  string  `json:"status"`
+	Accrual float64 `json:"accrual"`
+}
```

Comment:
Рекомендовал бы сгруппировать типы так, чтобы типы и их методы были ближе друг к другу. 
```suggestion
type ResponseAccrual struct {
	Order   string  `json:"order"`
	Status  string  `json:"status"`
	Accrual float64 `json:"accrual"`
}

type AccrualClient struct {
	serviceURL string
	logger     *logger.Logger
}

func NewAccrualClient(serviceURL string, logger *logger.Logger) *AccrualClient {
	instance := &AccrualClient{
		serviceURL: serviceURL,
		logger:     logger,
	}
	return instance
}
```